controllers/designate_controller.go

```diff
@@ -1059,9 +1059,54 @@ func (r *DesignateReconciler) generateServiceConfigMaps(
 	// - %-config configmap holding minimal designate config required to get the service up, user can add additional files to be added to the service
 	// - parameters which has passwords gets added from the ospSecret via the init container
 	//
+	Log := r.GetLogger(ctx)
 
 	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(designate.ServiceName), map[string]string{})
+	replicas := int(*instance.Spec.DesignateBackendbind9.Replicas)
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      designate.DesignateBindKeySecret,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	// Fetch the secret, create it if not found
+	_, err := controllerutil.CreateOrUpdate(ctx, h.GetClient(), secret, func() error {
```

Comment:
Why don't you use EnsureSecret for creating it?
https://github.com/openstack-k8s-operators/lib-common/blob/814ac3823abe9baa929a74cc8e03086d650f5981/modules/common/secret/secret.go#L281-L286

https://github.com/search?q=repo%3Aopenstack-k8s-operators%2Fdesignate-operator%20EnsureSecrets&type=code

Reply:
Note that EnsureSecret aslo updates the hash that can be used to recreate the pods when the secret is updated.

Reply:
It could be done that way. You first GetSecret.. if it doesn't exist, create the stringData map, create util.Template with the CustomData (I think) set to the that map and call EnsureSecrets with that.. If does exist, you do the work that's implemented in the mutator, getting the original contents of the secret first and creating/omitting entries as required and then you call EnsureSecrets. In some respects what Omer had done here is simpler/more straightforward. The only thing missing is to keep a handle of the secret and call secret.Hash() on it.